packages/kbn-kibana-json-schema/src/kibana_json_schema.ts

```diff
@@ -0,0 +1,145 @@
+      type: 'string',
+      pattern: '^(kibana|v?\\d+(\\.\\d+){0,2})$',
+    },
+    kibanaVersion: {
+      description: dedent`
+        The version of Kibana the plugin is compatible with, defaults to "version".
+      `,
+      type: 'string',
+      pattern: '^(kibana|)$',
+    },
+    type: {
+      description: 'Type of the plugin, defaults to `standard`.',
+      enum: ['standard', 'preboot'],
+    },
+    configPath: {
+      description:
+        'Root configuration path used by the plugin, defaults to "id" in snake_case format.',
+      oneOf: [
+        { type: 'string' },
+        {
+          type: 'array',
+          items: { type: 'string' },
+        },
+      ],
+    },
+    requiredPlugins: {
+      description: dedent`
+        An optional list of the other plugins that MUST BE installed and enabled for this
+        plugin to function properly.
+      `,
+      type: 'array',
+      items: { type: 'string' },
+    },
+    optionalPlugins: {
+      description: dedent`
+        An optional list of the other plugins that if installed and enabled **may be**
+        leveraged by this plugin for some additional functionality but otherwise are
+        not required for this plugin to work properly.
+      `,
+      type: 'array',
+      items: { type: 'string' },
+    },
+    requiredBundles: {
+      description: dedent`
+        An optional list of the other plugins that if installed and enabled MAY BE leveraged
+        by this plugin for some additional functionality but otherwise are not required for
+        this plugin to work properly.
+
+        The plugins listed here will be loaded in the browser, even if the plugin is
+        disabled. Required by \`@kbn/optimizer\` to support cross-plugin imports.
+        "core" and plugins already listed in \`requiredPlugins\` do not need to be
+        duplicated here.
+      `,
+      type: 'array',
+      items: { type: 'string' },
+    },
+    ui: {
+      description: dedent`
+        Specifies whether plugin includes some client/browser specific functionality
+        that should be included into client bundle via \`public/ui_plugin.js\` file.
+      `,
+      type: 'boolean',
+    },
+    server: {
+      description: dedent`
+        Specifies whether plugin includes some server-side specific functionality.
+      `,
+      type: 'boolean',
+    },
+    extraPublicDirs: {
+      description: dedent`
+        Specifies directory names that can be imported by other ui-plugins built
+        using the same instance of the @kbn/optimizer. A temporary measure we plan
+        to replace with better mechanisms for sharing static code between plugins
+        @deprecated To be deleted when https://github.com/elastic/kibana/issues/101948 is done.
+      `,
+      type: 'array',
+      items: { type: 'string' },
+    },
+    serviceFolders: {
+      description: dedent`
+        Only used for the automatically generated API documentation. Specifying service
```

Comment:
nit - newlines seem to be preserved causing the tooltips to wrap early

Reply:
Does it look bad in VSCode for you? It looked fine to my eye, and seemed best to keep things readable here.

Reply:
nope all good